rocketpool/node/collectors/governance-collector.go

```diff
@@ -0,0 +1,105 @@
+package collectors
+
+import (
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/rocket-pool/smartnode/bindings/dao/protocol"
+	"github.com/rocket-pool/smartnode/bindings/rocketpool"
+	"github.com/rocket-pool/smartnode/bindings/types"
+	"golang.org/x/sync/errgroup"
+)
+
+// Represents the collector for onchain governance metrics
+type GovernanceCollector struct {
+	// the number of active onchain proposals pending
+	onchainPending *prometheus.Desc
+
+	// the number of active onchain proposals in Phase 1
+	onchainPhase1 *prometheus.Desc
+
+	// the number of active onchain proposals in Phase 2
+	onchainPhase2 *prometheus.Desc
+
+	// The Rocket Pool Contract manager
+	rp *rocketpool.RocketPool
+
+	// Prefix for logging
+	logPrefix string
+}
+
+// Create a new SnapshotCollector instance
+func NewGovernanceCollector(rp *rocketpool.RocketPool) *GovernanceCollector {
+	subsystem := "governance"
+	return &GovernanceCollector{
+		onchainPending: prometheus.NewDesc(prometheus.BuildFQName(namespace, subsystem, "onchain_pending"),
+			"The number of pending onchain proposals",
+			nil, nil,
+		),
+		onchainPhase1: prometheus.NewDesc(prometheus.BuildFQName(namespace, subsystem, "onchain_phase1"),
+			"The number of onchain proposals in Phase 1",
+			nil, nil,
+		),
+		onchainPhase2: prometheus.NewDesc(prometheus.BuildFQName(namespace, subsystem, "onchain_phase2"),
+			"The number of onchain proposals in Phase 2",
+			nil, nil,
+		),
+		rp:        rp,
+		logPrefix: "Governance Collector",
+	}
+}
+
+// Write metric descriptions to the Prometheus channel
+func (collector *GovernanceCollector) Describe(channel chan<- *prometheus.Desc) {
+	channel <- collector.onchainPending
+	channel <- collector.onchainPhase1
+	channel <- collector.onchainPhase2
+}
+
+// Collect the latest metric values and pass them to Prometheus
+func (collector *GovernanceCollector) Collect(channel chan<- prometheus.Metric) {
+
+	// Sync
+	var wg errgroup.Group
+	var err error
+	var onchainProposals []protocol.ProtocolDaoProposalDetails
+	onchainPending := float64(0)
+	onchainPhase1 := float64(0)
+	onchainPhase2 := float64(0)
+
+	// Get onchain proposals
+	wg.Go(func() error {
+		onchainProposals, err = protocol.GetProposals(collector.rp, nil)
+		if err != nil {
+			return fmt.Errorf("error fetching onchain proposals: %w", err)
+		}
+		for _, proposal := range onchainProposals {
+			if proposal.State == types.ProtocolDaoProposalState_Pending {
+				onchainPending += 1
+			} else if proposal.State == types.ProtocolDaoProposalState_ActivePhase1 {
+				onchainPhase1 += 1
+			} else if proposal.State == types.ProtocolDaoProposalState_ActivePhase2 {
+				onchainPhase2 += 1
+			}
```

Comment:
```suggestion
			} else {
				onchainClosed += 1
```
When we range over `onchainProposals` , any `onchainProposals` that isn't a `types.ProtocolDaoProposalState_Pending, types.ProtocolDaoProposalState_ActivePhase1, or types.ProtocolDaoProposalState_ActivePhase2` can land in the `onchainClosed` bucket.